metadata/dashboard/dashboard_metadata.go

```diff
@@ -1074,16 +1121,219 @@ func (m *MetadataServer) getSourceDataIterator(name, variant string, limit int64
 	return primary.IterateSegment(limit)
 }
 
+type VariantDuck interface {
```

Comment:
whats a duck?

Reply:
just a reference to duck typing, I wasn't sure what to call the interface to also denote it's sole purpose is to just be a common result for all variants. maybe `VariantResult`?

Reply:
swapped to `VariantResult`